codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/middleware/OperationEndpointResolverMiddleware.kt

```diff
@@ -61,8 +61,8 @@ class OperationEndpointResolverMiddleware(
     ) {
         val output = MiddlewareShapeUtils.outputSymbol(ctx.symbolProvider, ctx.model, op)
         writer.write(
-            "\$N<\$N>(endpointResolver: config.endpointResolver, endpointParams: endpointParams)",
-            EndpointTypes.EndpointResolverMiddleware,
+            "\$N<\$N, EndpointParams>(endpointResolverBlock: { [config] in try config.endpointResolver.resolve(params: \$\$0) }, endpointParams: endpointParams)",
+            endpointResolverMiddlewareSymbol,
```

Comment:
The type of endpoint resolver to be created (smithy or AWS) is configurable.

Also, `endpointResolverBlock` is passed in so that the middleware can resolve the endpoint without knowing anything about the type of `endpointResolver`.  This is needed due to limitations of the Swift type system in supported versions (i.e. cannot resolve protocols with associated types at runtime.)